### PR TITLE
Add PeerConnection.connectionState.

### PR DESCRIPTION
**Pull request by pthatcherg**
As a possible solution to https://github.com/w3c/webrtc-pc/issues/278

### CODE REVIEW

**Comment by juberti** [comment]
My initial sense is that this is probably a mistake. I think we should just make the existing top-level state reflect the overall 'connection state' (e.g. ICE + DTLS, perhaps with a rename), and if you want more data, poll the individual IceTransport/DtlsTransport objects.

**Comment by pthatcherg** [comment]
FYI, I've updated the PR to be ICE+DTLS instead of just DTLS (It's not
pc.connectionState rather than pc.dtlsState).

**Comment by juberti** [comment]
As such, I think this should replace .iceState rather than adding a new state.

**Comment by pthatcherg** [comment]
I think that makes sense.  It shouldn't be hard to polyfill either the old one (.iceConnectionState) or the new one (.connectionState) once DtlsTransport and IceTransport are implemented.

**Comment by martinthomson** [comment]
From Seattle: a "new" state will be added, but this is otherwise OK.

File: webrtc.html [hunk 1]
```diff
@@ -1561,6 +1574,18 @@
...
+
+          <dt>attribute EventHandler onconnectionstatechange</dt>
+
+          <dd>
+            This event handler, of event handler event type <code><a href=
+            "#event-connectionstatechange">connectionstatechange</a></code>,
+            MUST be fired by all objects implementing the
+            <code><a>RTCPeerConnection</a></code> interface. It is
+            called any time the <a href=
+            "#dom-peerconnection-connection-state"><code>RTCPeerConnection</code>.connectionstate</a>
```
Comment (martinthomson) [inline webrtc.html]: Check case.

Comment (alvestrand) [inline webrtc.html]: the attribute name should be connectionState, not connnectionstate, I think.

**Comment by alvestrand** [comment]
Bernard will update.

File: webrtc.html (hunk 1 above)
Comment (pthatcherg) [inline webrtc.html]: Fixed the case.

File: webrtc.html [hunk 2]
```diff
@@ -1558,6 +1571,18 @@
...
+
+          <dt>attribute EventHandler onconnectionstatechange</dt>
+
+          <dd>
+            This event handler, of event handler event type <code><a href=
+            "#event-connectionstatechange">connectionstatechange</a></code>,
+            MUST be fired by all objects implementing the
+            <code><a>RTCPeerConnection</a></code> interface. It is
+            called any time the <a href=
+            "#dom-peerconnection-connection-state"><code>RTCPeerConnection</code>.connectionState</a>
```
Comment (adam-be) [inline webrtc.html]: I will do a PR that removes all the "it is called when" text from EventHandler attribute descriptions. But that shouldn't stop this PR from landing.

**Pull request merged**